swabbie-core/src/main/kotlin/com/netflix/spinnaker/config/SwabbieProperties.kt

```diff
@@ -61,8 +61,22 @@ class CloudProviderConfiguration {
   }
 }
 
+/**
+ * Notifications have both an [enabled] and a [required] configuration.
+ *  [enabled] is whether or not the notification agent runs for this configuration.
+ *  [required] is whether or not a resource must be notified before it can be deleted.
+ *
+ * If [enabled] is true and [required] is true then notifications will be sent to the owner.
+ *
+ * If [enabled] is true and [required] is false a resource will be marked as 'notification not needed'
+ *  and will be able to be deleted. No notifications will be sent.
+ *
+ * If [enabled] is false then the notification agent will not run for this configuration
+ *  and resources of this type won't be able to be deleted.
```

Comment:
I'd leave a TODO here for removing the hard dependency between resource and notification agents but think this PR makes sense as an interim way forward.

Reply:
Ok will add. Thanks!